src/Shared/DiagnosticIds/Experiments.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma warning disable CA1716
+namespace Microsoft.Shared.DiagnosticIds;
+#pragma warning restore CA1716
+
+/// <summary>
+/// Experiments supported by this repo.
+/// </summary>
+/// <remarks>
+/// When adding a new experiment, add a corresponding suppression to the root <c>Directory.Build.targets</c> file.
+/// </remarks>
+internal static class Experiments
+{
+#pragma warning disable S1075 // URIs should not be hardcoded
+    internal const string UrlFormat = "https://aka.ms/dotnet-extensions-warnings/{0}";
+#pragma warning restore S1075 // URIs should not be hardcoded
+
+    internal const string Resilience = "EXTEXP0001";
+    internal const string Compliance = "EXTEXP0002";
+    internal const string Telemetry = "EXTEXP0003";
+    internal const string TimeProvider = "EXTEXP0004";
+    internal const string AutoClient = "EXTEXP0005";
+    internal const string AsyncState = "EXTEXP0006";
+    internal const string HealthChecks = "EXTEXP0007";
+    internal const string ResourceMonitoring = "EXTEXP0008";
+    internal const string Hosting = "EXTEXP0009";
+    internal const string ObjectPool = "EXTEXP0010";
+    internal const string DocumentDb = "EXTEXP0011";
```

Comment:
We also need a doc listing all current IDs and explaining which API is each of these IDs light up, if enabled. 
The doc should also give guidance on what to do if the ID is encountered - for example, dotnet/aspnetcore has already encountered the TBD warning in the past (https://github.com/dotnet/aspnetcore/pull/49297#issuecomment-1629154246), so once these new IDs flow through what should the maintainers of the repo do. Is it safe to suppress, or should dotnet/aspnetcore rewrite something?